core/ir/opnd.h

```diff
@@ -1535,6 +1537,15 @@ DR_API
 opnd_t
 opnd_create_immed_float(float f);
 
+DR_API
+/**
+ * Returns an immediate double operand with value \p d.
+ * The caller's code should use proc_save_fpstate() or be inside a
+ * clean call that has requested to preserve the floating-point state.
```

Comment:
I know the float one says this, but it is a little stale these days: for x86 we always save the SIMD state and only don't auto-save the legacy x87 state.  For AArchXX aren't we also always saving any floating-point state?  So I would say tweak the docs to make that part conditional on legacy x87 floating point (on x86 modern compilers almost always used the SIMD floating point support and not the old x87).

Reply:
> For AArchXX aren't we also always saving any floating-point state?

It's not clear from this issue whether that is the case: https://github.com/DynamoRIO/dynamorio/issues/2629